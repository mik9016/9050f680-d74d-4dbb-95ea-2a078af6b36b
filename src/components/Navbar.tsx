import React, { FC, ReactElement, useState, useContext } from 'react'
import { Link } from 'react-router-dom'
import { Box, Container, Toolbar } from '@mui/material'
import SearchIcon from '@mui/icons-material/Search'
import { GlobalContext } from '../contexts/Context'

import { IconTextField } from './IconTextField'
import { ShoppingCart } from './ShoppingCart'

const Navbar = (): React.ReactElement => {
  const globalCtx = useContext(GlobalContext)

  return (
    <Box
      sx={{
        width: '100%',
        height: 'auto',
        padding: 1,
        paddingLeft: 3,
        backgroundColor: 'primary.main',
        position: 'fixed',
        zIndex: '4',
      }}
    >
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <IconTextField label="Search" iconStart={<SearchIcon />} />
          <Box sx={{ flexGrow: 1, display: 'flex' }}>
            <Box
              sx={{
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'flex-end',
                alignItems: 'center',
                marginLeft: 'auto',
                paddingLeft: '1rem',
              }}
            >
              <Link to={'/cart'}>
                <ShoppingCart number={globalCtx.currentCartNumber} />
              </Link>
            </Box>
          </Box>
        </Toolbar>
      </Container>
    </Box>
  )
}

export default Navbar
