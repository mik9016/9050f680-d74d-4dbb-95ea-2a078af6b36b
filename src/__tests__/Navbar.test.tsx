import React from 'react'

import { act, create } from 'react-test-renderer'

import Navbar from '../components/Navbar'
import { BrowserRouter as Router } from 'react-router-dom'

test(`renders Navbar without crashing`, async () => {
  const root = create(
    <Router>
      <Navbar />
    </Router>
  )

  expect(root.toJSON()).toMatchSnapshot()

  act(() => {
    root.unmount()
  })
})
