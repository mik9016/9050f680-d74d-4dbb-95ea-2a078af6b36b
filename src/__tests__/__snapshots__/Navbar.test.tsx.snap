// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar without crashing 1`] = `
<div
  className="MuiBox-root css-1h1t0bo"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Search
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1in441m"
            >
              <span>
                Search
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiBox-root css-1l0r181"
      >
        <div
          className="MuiBox-root css-df8xth"
        >
          <a
            href="/cart"
            onClick={[Function]}
          >
            <span
              className="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
                data-testid="ShoppingCartOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError BaseBadge-badge BaseBadge-invisible css-zkqb44-MuiBadge-badge"
              />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
